hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,89 @@
+- Assuming a continuously-live entropy source, each instance can also optionally be used as a non-determinstic TRNG (true random number generator, also called a non-deterministic random bit generator or NRBG in SP 800-90C).
+    - In this mode, an instance also meets the requirements laid out for a PTG.3 class RNG, the strongest class laid out in AIS31.
+    - Implementation follows the NRBG "Oversampling Construction" approved by SP 800-90C, to meet both CC and FIPS TRNG constructions.
+- In addition to the approved DRNG mode, any instance can also operate in "Fully Deterministic mode", meaning the seed depends entirely on application inputs or personalization strings.
+    - This provides an approved means of key-derivation in FIPS 140-2.
+
+## Description
+
+Though the recommendations in AIS31 are based around broad functional requirements, the recommendations in SP 800-90 are very prescriptive in nature, outlining the exact constructs needed for approval.
+Thus the interface and implementation are largely driven by these explicit constructs, particularly the CTR_DBRG construct.
+
+
+The CSRNG IP consists of four main components:
+1. An AES primitive
+2. The CTR_DRBG state-machine (CTR_DRBG_FSM) which drives the AES primitive, performing the various encryption sequences proscribed in SP 800-90A.
+These include:
+
+    1. **The Derivation Function:**  
+       Part of the instantiation and reseed routines, this routine assembles the previous seed material (on reseed only), application inputs, and entropy.
+    2. **The Instantiation Routine:**
+       Combines application inputs, external entropy and nonce (more entropy) via the derivation function.
+    3. **The Reseed Routine:**
+       Combines the previous seed material with external entropy to generate a new seed.
+    4. **The Generate Routine:**
+       Generates up to CSRNG_MAX_GENERATE random bits.
+       If called with prediction_resistance_flag, forces a reseed.
+     5. **The Update Routine:**
+        Updates the internal state of the DRBG instance after each generate call.
+3. State vectors for each DRNG instance.
+4. Interface logic and access control for each instance.
+
+## Note on the term "Entropy"
+
+Every DRNG requires some initial seed material, and the requirements for the generation of that seed material varies greatly between standards, and potentially between CC security targets.
+In all standards considered, DRNG's require some "entropy" from an external source to create the initial seed.
+However, the rules for obtaining said entropy differ.
+Furthermore the required delivery mechanisms differ.
+For this reason we must make a clear distinction between "Physical" (or "Live" or "True") entropy and "Factory Entropy".
+This distinction is most important when considering the creation of IP which is both compatible with  IPS 140-3 recommendations, and FIPS 140-2 use cases (which have already been certified for a variety of applications).
```

Comment:
Also, do you have a link to the specific FIPS 140-3 recommendations you are meaning here?

Reply:
It's just part of SP 800-90, (which are cited in 140-3), but I'll say that explicitly.